BraintreeCore/src/main/java/com/braintreepayments/api/InvalidArgumentException.kt

```diff
@@ -0,0 +1,9 @@
+package com.braintreepayments.api
+
+import java.lang.Exception
+
+/**
+ * Error thrown when arguments provided to a method are invalid.
+ * @param message the error message
+ */
+class InvalidArgumentException internal constructor(message: String?) : Exception(message)
```

Comment:
```suggestion
open class InvalidArgumentException internal constructor(message: String?) : Exception(message)
```

Reply:
Yeah technically though to avoid a breaking change it's necessary. If a merchant can do it now they need to be able to subclass.

This will be true for all public facing API since we never marked anything `final` on the Java side.

Reply:
I wish there was a better way. We'll have to next major version remove `open` in a lot of places once we get to `v5`.

Reply:
i havent done java in a while, but if the default access modifier is `package-private`, then merchants never had access to this class. So we shouldnt be introducing any breaking change if we left out `open`. Or am i missing something?

Reply:
Kotlin classes are `public final` by default when used in Java. Since we marked [InvalidArgumentException.java](https://github.com/braintree/braintree_android/blob/master/BraintreeCore/src/main/java/com/braintreepayments/api/InvalidArgumentException.java) public in Java, we'll need to make it `open` on the Kotlin side because it isn't marked `final`.

Historically we've stayed away from finalizers in the Android SDK, mostly because mockito blows up unless you configure `mock-maker-inline` and all that stuff. Even then, when you add in powermock and Robolectric the unit tests become flakey and unpredictable.